R/init.R

```diff
@@ -208,7 +208,12 @@ init <- function(data,
   ## `data` - `modules`
   if (inherits(data, "teal_data")) {
     if (length(teal_data_datanames(data)) == 0) {
-      stop("`data` object has no datanames and its environment is empty. Specify `datanames(data)` and try again.")
+      stop(
+        paste0(
+          "`data` object has no datanames and its environment is empty.\n",
+          "Specify `datanames(data)` or make sure that the `data` object is not empty."
+        )
+      )
```

Comment:
This change changes nothing.
Also, don't use `paste` in condition messages unless you have to.

Reply:
I will remove the paste.
I thought the main issue was that we're computing the datanames so nudging the user when it's empty was the way to do it. What did you have in mind?

Reply:
The line you appended repeats the previous one and adds nothing.

Reply:
```suggestion
      stop(
          "`data` object has no datanames and its environment is empty.\n",
          "Specify `datanames(data)` or make sure that the `data` object is not empty."
      )
```

Reply:
Then I don't get the crux of the issue yet.
Was it not to let the user know that the `data` slot is empty which they should fix, which was missing earlier?